Sources/Testing/Test.Case.Generator.swift

```diff
@@ -93,34 +117,124 @@ extension Test.Case {
+            try await testFunction(element)
+          }
+        }
+      } else {
+        self.init(sequence: sequence) { element in
+          Test.Case(values: [element], parameters: parameters) {
+            try await testFunction(element)
+          }
+        }
+      }
+    }
+
+    /// Initialize an instance of this type that iterates over the specified
+    /// collection of 2-tuple argument values.
+    ///
+    /// - Parameters:
+    ///   - collection: The collection of 2-tuple argument values for which test
+    ///     cases should be generated.
+    ///   - parameters: The parameters of the test function for which test cases
+    ///     should be generated.
+    ///   - testFunction: The test function to which each generated test case
+    ///     passes an argument value from `collection`.
+    ///
+    /// @Comment {
+    ///   - Bug: The testing library should support variadic generics.
+    ///     ([103416861](rdar://103416861))
+    /// }
+    init<E1, E2>(
+      arguments collection: S,
+      parameters: [Test.ParameterInfo],
+      testFunction: @escaping @Sendable ((E1, E2)) async throws -> Void
+    ) where S: Collection, S.Element == (E1, E2) {
+      self.init(sequence: collection, parameters: parameters, testFunction: testFunction)
+    }
+
     /// Initialize an instance of this type that iterates over the specified
     /// zipped sequence of argument values.
     ///
     /// - Parameters:
     ///   - zippedCollections: A zipped sequence of argument values for which
     ///     test cases should be generated.
+    ///   - parameters: The parameters of the test function for which test cases
+    ///     should be generated.
     ///   - testFunction: The test function to which each generated test case
     ///     passes an argument value from `zippedCollections`.
     init<C1, C2>(
       arguments zippedCollections: Zip2Sequence<C1, C2>,
+      parameters: [Test.ParameterInfo],
       testFunction: @escaping @Sendable ((C1.Element, C2.Element)) async throws -> Void
-    ) where S == Zip2Sequence<C1, C2> {
-      self.init(sequence: zippedCollections) { element in
-        Test.Case(arguments: [element]) {
-          try await testFunction(element)
+    ) where S == Zip2Sequence<C1, C2>, C1: Collection, C2: Collection {
+      self.init(sequence: zippedCollections, parameters: parameters, testFunction: testFunction)
+    }
+
+    /// Initialize an instance of this type that iterates over the specified
+    /// dictionary of argument values.
+    ///
+    /// - Parameters:
+    ///   - dictionary: A dictionary of argument values for which test cases
+    ///     should be generated.
+    ///   - parameters: The parameters of the test function for which test cases
+    ///     should be generated.
+    ///   - testFunction: The test function to which each generated test case
+    ///     passes an argument value from `dictionary`.
+    init<Key, Value>(
```

Comment:
Please document why this overload is useful (perf optimization for a perceived common case.)